packages/app/src/containers/LayerStyleInspector.tsx

```diff
@@ -0,0 +1,302 @@
+import { Divider, InputField, Spacer, Slider } from 'noya-designsystem';
+import { Selectors } from 'noya-state';
+import { memo, useCallback, useMemo, ReactNode } from 'react';
+import type FileFormat from '@sketch-hq/sketch-file-format-ts';
+import ArrayController from '../components/inspector/ArrayController';
+import {
+  useApplicationState,
+  useSelector,
+} from '../contexts/ApplicationStateContext';
+import * as InspectorPrimitives from '../components/inspector/InspectorPrimitives';
+import useShallowArray from '../hooks/useShallowArray';
+import getMultiValue from '../utils/getMultiValue';
+import FillRow from '../components/inspector/FillRow';
+import BorderRow from '../components/inspector/BorderRow';
+import ShadowRow from '../components/inspector/ShadowRow';
+import withSeparatorElements from '../utils/withSeparatorElements';
+import NameInspector from './NameInspector';
+
+const StyleOpacityInspector = memo(function OpacityInspector() {
```

Comment:
Oh! I was about to do that, join them but we need to change this lines in the inspector. 
![image](https://user-images.githubusercontent.com/16924758/113457224-dd4b9500-93d4-11eb-891c-957b34d99137.png)


By either sending them the selectors before or conditionally selecting. That's the only difference. the selector

Reply:
Something similiar to how NameInspector would work, where different types of objects can interact with it. So the selector shouldn't be inside the inpector I think

Reply:
Ah, right. I can think of 2 options here:

1. We pass an array of `styles` or style ids to `OpacityInspector` as a prop, and move the selector outside the `OpacityInspector`
2. We use a different selector within `OpacityInspector` that returns an array of styles, and depending on the current tab it will either return layer styles or shared styles

I like option 1 longer term, but I think option 2 would be a much smaller change, and more consistent with the rest of this PR, since it will mirror how we currently handle actions. Something like...

```ts
// pseudocode, not sure if these functions all exist
export const getSelectedStyles = (state: ApplicationState): Sketch.Style[] => {
  const currentTab = getCurrentTab(state);
  return currentTab === 'canvas' 
    ? getSelectedLayers(state).flatMap(layer => layer.style ? [layer] : []) 
    : getSharedStyles(state).map(style => style.value)
};
```

What do you think?

Reply:
I think option 2 it's better, I just added that one. We would have to change how NameInspector works to match it maybe but I just changed it. It looks cleaner and it follows a better pattern .